test/gui/shared/steps/steps.py

```diff
@@ -1103,4 +1155,53 @@ def step(context, resource):
 )
 def step(context, publicLinkName, password):
     publicLinkDialog = PublicLinkDialog()
-    publicLinkDialog.changePassword(publicLinkName, password)
+    publicLinkDialog.verifyPublicLinkName(publicLinkName)
+    publicLinkDialog.changePassword(password)
+
+
+@When('the user searches for collaborator "|any|" using the client-UI')
+def step(context, collaborator):
+    shareItem = SharingDialog()
+    shareItem.searchCollaborator(collaborator)
+
+
+@When(
+    'the user searches for collaborator with autocomplete characters "|any|" using the client-UI'
+)
+def step(context, collaborator):
+    shareItem = SharingDialog()
+    shareItem.searchCollaborator(collaborator)
```

Comment:
How come these 2 step definitions has the same logic? 🤔 And if it is the same logic, then we can reuse the steps.

Reply:
Because when you type the full collaborator name or 3 characters, it is just the same (you search by typing)

Reply:
You can put this logic in a function and then call the function from both of these step definitions. And if the logic should be changed someday, it is easier if we reuse the code.
```
shareItem = SharingDialog()
shareItem.searchCollaborator(collaborator)
```